src/quicklist.c

```diff
@@ -476,22 +475,8 @@ REDIS_STATIC int _quicklistNodeAllowInsert(const quicklistNode *node,
     if (unlikely(QL_NODE_IS_PLAIN(node) || isLargeElement(sz)))
         return 0;
 
-    size_t lp_overhead = sz;
-    /* size of encode offset */
-    if (sz < 64) lp_overhead += 1;
-    else if (sz < 4096) lp_overhead += 2;
-    else lp_overhead += 5;
-
-    /* size of backlen offset */
-    if (lp_overhead <= 127) lp_overhead += 1;
-    else if (lp_overhead < 16383) lp_overhead += 2;
-    else if (lp_overhead < 2097151) lp_overhead += 3;
-    else if (lp_overhead < 268435455) lp_overhead += 4;
-    else lp_overhead += 5;
-
-    /* new_sz overestimates if 'sz' encodes to an integer type */
-    size_t new_sz = node->sz + lp_overhead;
-    assert(new_sz > lp_overhead); /* Catch size_t overflow */
+
+    size_t new_sz = node->sz + sz;
```

Comment:
let's add a comment stating that we ignore the size of the headers of the new record since they'll be about 10 bytes at most.

maybe also mention that we're not concerned about overflows because the above plain node / large element check.
and that the effect of this 10 bytes nuance is not a real concern for _quicklistNodeSizeMeetsOptimizationRequirement and sizeMeetsSafetyLimit since they're mostly a size guideline than a real hard limit.

Reply:
actually, another look at _quicklistNodeSizeMeetsOptimizationRequirement.
considering a default setting of 8192, if we'll create a long quickly and will breach the limit of all nodes by a few bytes (let's say have 8198), we'll constantly create a large internal fragmentation overhead.
so on a second thought, maybe we do need a proper estimation of the header (or at least an over estimation)?

WDYT?

Reply:
I set the overhead to 8, which ensures that it is safe even if the fill is -5, which will also result in a maximum of 6(8-2) bytes of wasted space.
Also, I have tried adding a method to the listpack to accurately calculate the entry size, but it would cause a small performance penalty because we would need to calculate the entry size twice (another in lpInsert).